### PR TITLE
Update Payments menu item display and linking behavior

### LINKED ISSUE #51111: [Enhancement]: Decouple WooPayments incentive and Payments nav item
### Describe the solution you'd like

Currently, the Payments nav item notification badge in wp-admin is tied to WooPayments incentive. The badge is not visible if an incentive is unavailable in the merchant's country. We need to change this behavior.

#### Acceptance criteria:

* When the WooPayments plugin is NOT installed, regardless of an incentive being available - the left nav payments item will always show with the notification badge.
* If an incentive is available, clicking the nav item will take the merchant to the fake connect page.
* If an incentive is unavailable, clicking the nav item will take the merchant to the payments task page.
* Update the WooPayments tile on all surfaces to use the new payment methods shared component (#49300) so it is consistent with our other pages.
* Remove the recommended badge on Stripe when WooPayments is presented.

### Describe alternatives you've considered

_No response_

### Additional context

image

### PR DESCRIPTION
**Pull request by vladolaru**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We detach the Payments top-level menu item from the presence of an active WooPayments incentive:
- When the WooPayments plugin is NOT active, regardless of an incentive being available - the left nav payments item will always show.
- When the WooPayments plugin is active, it will take over the Payments menu item for its use.
- If an incentive is available, clicking the nav item will take the merchant to the incentive page.
- If an incentive is unavailable or dismissed, clicking the nav item will take the merchant to the currently active Payments task page (either the base "Get paid" task or the "Get paid with WooPayments" task).

Additionally, we remove the recommended treatment of PSP suggestions when WooPayments is recommended.

Here is a recording of the new behavior on a fresh JurassicNinja site: 
[Screenshot 2024-09-20 at 09 55 12](https://github.com/user-attachments/assets/981a551a-348f-4d43-a862-3778c65d0720)

Closes #51111

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JurassicNinja site with just WooCommerce on this PR's live branch (here is [a direct JN create link](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce=update/51111-payments-menu-item))
1. Go to the newly created site WP admin dashboard
1. Go through the WooCommerce profiler (click on the "WooCommerce" menu item if it doesn't start automatically), click on "Skip guided setup", and choose India as your store's location
1. Go to WooCommerce  > Home and you should see a "Get paid" task and a "Payments" top-level menu item with a red dot notice:
Screenshot 2024-09-20 at 10 01 21
1. Click on the "Payments" top-level menu item and you should see the Payments task page (notice the `/wp-admin/admin.php?page=wc-admin&task=payments` URL):
Screenshot 2024-09-20 at 10 04 06
1. Click "Enable" for the "Cash on delivery" offline payment method in the bottom "Offline payments methods" section.
1. Refresh the page, and the "Payments" top-level menu item should NOT have a red dot notice.
1. Go to WooCommerce > Home and the "Get paid" task should be complete (striked out).
1. Click on the "Payments" top-level menu item and click on "Get started" for the Stripe recommended payment provider:

1. After the Stripe extension is installed, go to WooCommerce > Settings > Payments and click on "Finish setup" for Stripe:

1. Once on the Stripe extensions settings page, click on "Create or connect a test account instead" and them "Skip this form":

1. Once the Stripe test account is set up and you are redirected back to the Stripe extension settings page, go to WooCommerce > Home and make sure the "Get paid" task is completed (striked out)
1. Click on the "Payments" top-level menu item and Stripe shouldn't be present:
Screenshot 2024-09-20 at 10 16 31
1. Go to WooCommerce > Settings > General and change your store's location country to United States (the specific state doesn't matter)
1. Go to WooCommerce > Home and you should see a completed "Get paid with WooPayments" task (striked out) and a "Set up additional payments options" task in the "Things to do next" list:

1. Click on the "Set up additional payments options" task and you should see a page with WooPayments at the top and additional payment options below (the "Payments" top-level menu item should be present and NOT have a red dot notice):
Screenshot 2024-09-20 at 10 21 46
1. Click on the "Payments" top-level menu item and you should remain on the same page (notice the `wp-admin/admin.php?page=wc-admin&task=payments` URL)
1. Click on "Get started" for WooPayments, and after the WooPayments plugin is installed and activated, click on "Connect":

1. You should be taken to the Jetpack connection screen (approve the connection) and then to the WooPayments wizard (click on Cancel):
Screenshot 2024-12-06 at 15 52 03
1. Go to WooCommerce > Home and click on the "Set up additional payments options" task in the "Things to do next" list. You should be taken to the task page with WooPayments featuring a "Finish setup" button:

1. Click on the "Finish setup" button, and you will be taken to the WooPayments wizard. Click on Cancel and you will be taken to the WooPayments Connect page.
1. Click on "Enable sandbox mode" in the "I'm setting up the store for someone else" bottom section. After your sandbox account is set up, you will be taken to the Payments > Overview page with a complete/enabled account

1. Go to Plugins > Installed plugins and deactivate both WooPayments and WooCommerce Stripe Gateway.
1. Go to WooCommerce > Settings > Payments and disable the "Cash on delivery" payment method
1. Go to WooCommerce > Home and you should see "Get paid with WooPayments" task with a discount badge. At the same time, you should see a "Payments" top-level menu item with a red dot notice:

1. Click on the "Payments" top-level menu item and you will see the WooPayments incentive page:

1. Dismiss the incentive by clicking on the "No thanks" link and then click on "Just dismiss WooPayments" in the modal.
1. You will be redirected to the WooCommerce > Home page with a "Get paid with WooPayments" task without a discount badge and a "Payments" top-level main menu item with a red dot notice:

1. Click on the "Payments" top-level main menu item and you will be taken to the task page (notice the `wp-admin/admin.php?page=wc-admin&task=woocommerce-payments` URL). WooPayments is present, and there is no recommended badge for the gateway in the "Other payment providers" section below:

1. Click on "Enable" for the "Cash on delivery" offline payment method and refresh the page. The "Payments" top-level main menu item should NOT have a red dot notice anymore.
1. Go to WooCommerce > Home and click on the "Get paid with WooPayments" task. WooPayments will be activated and you will be redirected to the WooPayments settings page since you already have a connected account.
1. Go to WooCommerce > Home and the "Get paid with WooPayments" task should be completed (striked out), and in the "Things to do next" task list, you should have a "Set up additional payment options" task and a "Set up real payments on your store" task.
1. Click on the "Set up additional payment options" task, and you should see a task page with an additional payment options section and a "Activate BNPL instantly on WooPayments" section:

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by anu-rock** [comment]
I just finished my testing. All instructions checked out fine 👍

I have a couple of questions: (not directly related to code changes in this PR, but not completely unrelated)

> 15. Go to WooCommerce > Home and you should see a completed "Get paid with WooPayments" task (striked out) and a "Set up additional payments options" task in the "Things to do next" list:

By this point, we've done nothing with WooPayments, so seeing "Get paid with WooPayments" written feels out of place. Shouldn't it be simply "Get paid"?

> 16. Click on the "Set up additional payments options" task and you should see a page with WooPayments at the top and additional payment options below (the "Payments" top-level menu item should be present and NOT have a red dot notice):

Just confirming - if a merchant was meeting certain pre-conditions, would they be shown the switch incentive here? Apologies for the dumb question, I'm fully aware of the switch incentive as a concept but I'm not sure about all its triggers.

> 26. Click on the "Payments" top-level menu item and you will see the WooPayments incentive page:

Again, just confirming - we are showing the action incentive now because the merchant doesn't have any other payment provider enabled?

Onward to code review.

**Comment by anu-rock** [comment]
Double-checking if this item (from the issue description) is also targeted in this PR:

> Update the WooPayments tile on all surfaces to use the new payment methods shared component (https://github.com/woocommerce/woocommerce/pull/49300) so it is consistent with our other pages.

WooPayments Onboarding Flows

File: plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/index.js
Comment (anu-rock) [inline plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/index.js:253]: The original requirement was about removing the recommended badge from just Stripe, but this code change will remove the badge from all payment providers (including PayPal). Is this intentional? cc @timmy5685

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php
Comment (anu-rock) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php:245]: You've likely already checked - is it possible to get the list of offline methods from an existing WooCommerce API?

File: plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php
Comment (anu-rock) [inline plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php:440]: Just for my knowledge - is this more efficient than using an array literal?

**Review by anu-rock — COMMENTED** [review]
The PR implements the main requirements well. But I've left a few comments. Happy to do another round of review after you've addressed/answered them.
___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ⚠️  I didn't notice any test-related changes. Do we not have existing tests for the flows related to the Payments nav item and tasks? |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Documentation  | Not applicable |
| Testing instructions  | 🥇 Top notch! |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| I followed the testing instructions.  | ✅ The Payments nav item was shown even in the absence of WooPayments incentive. Clicking on the nav item redirected to the correct page depending on the availability of an incentive. Stripe did not have a recommended badge when WooPayments was available. | More details in [this comment](https://github.com/woocommerce/woocommerce/pull/51541#issuecomment-2363282134) |

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php
Comment (vladolaru) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php:245]: We don't have such an API.

File: plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php
Comment (vladolaru) [reply plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php:440]: It is a requirement from the WC linter to use long array syntax, not the short one. I believe it is to be in alignment with WP. It doesn't affect performance either way.

File: plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/index.js
Comment (vladolaru) [reply plugins/woocommerce-admin/client/task-lists/fills/PaymentGatewaySuggestions/index.js:253]: Yes, it is intentional as the intent is not to confuse users with multiple recommendations (WooPayments and something else). Stripe is just the one that usually got that badge because it was the highest priority. I've confirmed with Tim that the flows are as intended.

**Comment by vladolaru** [comment]
> Double-checking if this item (from the issue description) is also targeted in this PR:
>
> Update the WooPayments tile on all surfaces to use the new payment methods shared component (https://github.com/woocommerce/woocommerce/pull/49300) so it is consistent with our other pages.

We are already using the new component as seen in the video and testing instructions screenshots.

**Comment by vladolaru** [comment]
> > 15. Go to WooCommerce > Home and you should see a completed "Get paid with WooPayments" task (striked out) and a "Set up additional payments options" task in the "Things to do next" list:

> By this point, we've done nothing with WooPayments, so seeing "Get paid with WooPayments" written feels out of place. Shouldn't it be simply "Get paid"?

That title comes from the active task. When the store country is a WooPayments eligible one, the WooPayments flavored task superseeds the base one. But that doesn't mean that completing the task means setting up WooPayments; it can be any gateway. 

It would be more confusing if the task kept switching titles depending on completion.

**Comment by vladolaru** [comment]
> > 16. Click on the "Set up additional payments options" task and you should see a page with WooPayments at the top and additional payment options below (the "Payments" top-level menu item should be present and NOT have a red dot notice):

> Just confirming - if a merchant was meeting certain pre-conditions, would they be shown the switch incentive here? Apologies for the dumb question, I'm fully aware of the switch incentive as a concept but I'm not sure about all its triggers.

We never did show incentives on the Payments task page. We have a dedicated page for that linked from the Payments menu item.

**Comment by vladolaru** [comment]
> > Click on the "Payments" top-level menu item and you will see the WooPayments incentive page:

> Again, just confirming - we are showing the action incentive now because the merchant doesn't have any other payment provider enabled?

Yes. The testing instructions only capture the Action incentive. The Switch incentive requires an older, active store. But for the purpose of the PR, it doesn't matter the type of incentive being active.

**Comment by anu-rock** [comment]
> But that doesn't mean that completing the task means setting up WooPayments; it can be any gateway.

That's exactly my point. In the current setup, enabling a payment provider other than WooPayments will also complete the task titled "Get paid with WooPayments", which sounds confusing. Or am I interpreting it incorrectly?

**Comment by vladolaru** [comment]
@anu-rock about tests, we do have some unit tests (there are [changes](https://github.com/woocommerce/woocommerce/pull/51541/files#diff-186b42f77dfb1a3a78350679e5359e42fe5d405cc7bcdf985fde1881a7885cd6R168) in the PR). About the Payments nav item, we don't have tests.

**Review by anu-rock — APPROVED** [review]
Approving after discussing the task title on Slack p1726833956432069/1726819035.236429-slack-C03KTTK2YMA.

**Comment by thekendog** [comment]
It appears this was included in 9.4.1. It's broken.  I now have a constant Payments menu item with an orange bubble.  Clicking it shows a blank page.  If I enable a payment method from the regular settings area, the orange bubble goes away on the menu item, but the page is still blank when I click it.

Why was this added?  WooCommerce already has 4 menu items in the admin section.  I don't really want another one to clutter things up.  There probably should be a way to remove this easily, once it's fixed.

**Comment by dcx15** [comment]
> Why was this added?

I assume it is just a classic case of Automattic wanting to upsell something.

Fun.

**Comment by thekendog** [comment]
> > Why was this added?
> 
> I assume it is just a classic case of Automattic wanting to upsell something.
> 
> Fun.

I understand then wanting to upsell something and make money.  I'm fine with that.  There just needs to be a way to remove this menu item once I dismiss the task.  That and it needed more thorough testing I guess.

**Comment by mrleemon** [comment]
Same here. This "Payments" tab appears even though I don't have WooPayments enabled. Clicking it takes me to a blank page. I would like an option to remove this.

**Comment by maddisondesigns** [comment]
This PR either needs to be reversed or a fix put in place ASAP, to remove the extra `Payments` top-level menu item (which shows as a blank page. This PR was obviously never tested properly before being merged.

**Comment by mattthomasphoto** [comment]
Awful that you can’t dismiss this WooPayments page as I just the official Woo Stripe for payments and now there’s this extra menu for no reason but “advertising”

Using the below hides the Payments menu - however CSS to hide something that shouldn't be there in the first place!

`// Hide WooCommerce Payments menu item with CSS
function hide_payments_menu_css() {
echo '<style>
 .toplevel_page_admin\\?page\\=wc-admin\\&task\\=payments {
display: none !important;
 }
</style>';
}
add_action('admin_head', 'hide_payments_menu_css');`

This needs sorting ASAP!

**Comment by vladolaru** [comment]
@maddisondesigns, could you help me with some details on how you are testing this and getting the blank page? An SSR of the test store would also come in handy. Thank you.

**Comment by thekendog** [comment]
https://github.com/woocommerce/woocommerce/issues/39668#issuecomment-2476498613

**Comment by dcx15** [comment]
This feels like such an odd change and the [issue it closes](https://github.com/woocommerce/woocommerce/issues/51111) doesn't mention anything about _why_ it was needed. Is the purpose simply to signpost the WooPayments service offered by Woo?

There are more people posting about the page being blank at https://wordpress.org/support/topic/payments-menu-after-upgrading-to-9-4-1-empty-page/ too

**Comment by richboos** [comment]
This really shouldn't appear as a top-level menu item in WordPress. Payment integrations aren't something that is frequently visited. Compare to other top-level entries:

- WooCommerce - It's own thing, obviously
- Products - products come and go, prices change, stock levels change, etc
- Marketing - coupons get created and rotated

But Payments? Why? Who is changing out payment providers weekly?

I think we know why, which would be what @dcx15 implied - it's an advertisement decision, not a usage-based decision.

**Comment by maddisondesigns** [comment]
@vladolaru All I'm doing is clicking on the Payments top-level menu item that has appeared since the most recent update. Here's a (very short) screencast showing that the page is blank after clicking on the menu item. This is on a live site, but I've also tested this on a couple of local development sites, and the exact same thing happens.

https://github.com/user-attachments/assets/941d821b-e46f-4f77-90ed-918eaf633aca


And no, there was no `No thanks` link to click on. The menu item just appeared after the most recent update and it's always been showing a blank screen.

**Comment by maddisondesigns** [comment]
@vladolaru As mentioned over in this issue [#52812](https://github.com/woocommerce/woocommerce/issues/52812), this "fix" that has been merged, has still not actually fixed the issue! I'm still seeing a Payments menu item in my dashboard, with WooCommerce 9.4.2 Instead of seeing a blank page when I click on it though, it now just redirects me to the Payments Tab in the WC Settings.

Screencast: https://share.zight.com/P8uKZvOP

I'm seeing the same thing on multiple sites. This whole menu item needs to be completely removed from the Dashboard!

**Pull request merged**
